### DISCUSSION TITLE
Why upstream.healthcheck_parent is route?

### DISCUSSION THREAD

**Discussion by panhow** [apache/apisix#5924] [Q&A] [answered]
why upstream.healthcheck_parent is route?

`checker:run` will run after the object changed, but when i update a `upstream`, `checker:clear()`,`checker:stop()` are on route, route did not change, then the `checker` still checks the old upstream nodes?

**Comment by juzhiyuan** [comment, marked as answer]
Hi @panhow, could you please provide more context for your question? Do you mean https://github.com/apache/apisix/blob/master/apisix/upstream.lua#L64 ?

**Reply by panhow** [reply]
already got answer in https://github.com/apache/apisix/issues/5935#issuecomment-1001403983
thx~
